### PR TITLE
feat(logger): add `link`

### LINKED ISSUE #462: feat: Add `link` ansi wrapper
**Description**

Add an ANSI wrapper that enables hyperlinks/links in the terminal. More info at this [Gist](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda) on how to achieve the desired effect.

I've done this in one of my projects, but I think it might server others as well.

**Requirements**

- [ ] API to "wrap" a string with a link for stdout.

**Additional Context**

An example API:

```dart
Future<void> main() async {
  Logger(level: Level.verbose).info(
    linkWrap(
      'You can find issues here',
      'https://github.com/felangel/mason/issues/',
    ),
  );
}
```
```dart
String linkWrap(String message, String link) {
  final uri = '\x1B]8;;$link\x1B\\';
  final text = '$message\x1B]8;;\x1B\\';
  return '$uri$text';
}
```
hyperlink

### PR DESCRIPTION
**Pull request by Luckey-Elijah**
## Status
**READY**

## Description

Resolves #462

- Adds the `link` function for utility.
- ~Adds the `Logger.link`method to write links to `stdout` (or with option "writer")~

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/mason_logger/lib/src/link_wrap.dart [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/// Helper function for "wrapping" an message optional message with link.
```
Comment (Luckey-Elijah) [inline packages/mason_logger/lib/src/link_wrap.dart]: fix typo

File: packages/mason_logger/test/src/link_wrap_test.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        final output = linkWrap(message, uri);
+        final matcher = stringContainsInOrder(
+          [lead, '$uri', trail, message, lead, trail],
+        );
+
+        expect(output, matcher);
+      },
+    );
+
+    test('builds String with Uri when message is null: ', () {
```
Comment (Luckey-Elijah) [inline packages/mason_logger/test/src/link_wrap_test.dart:23]: When `message` is `null` it will just use the `uri`.

**Comment by Luckey-Elijah** [comment]
@felangel ready for review 🙂

File: packages/mason_logger/lib/src/mason_logger.dart [hunk 3]
```diff
@@ -49,6 +52,21 @@ class Logger {
     _stdout.writeln(message);
   }
 
+  /// Writes a message with a clickable link.
+  /// By default writes to `stdout` with [info], but any [StdoutWriter] works.
+  ///
+  /// **NOTE**: Only works on supported terminal emulators. Please see this
+  /// [link on what is supported](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda#supporting-apps).
+  void link(
```
Comment (felangel) [inline packages/mason_logger/lib/src/mason_logger.dart:60]: Do we need this? Can't we just interpolate `linkWrap` into existing messages?

Comment (Luckey-Elijah) [reply packages/mason_logger/lib/src/mason_logger.dart:60]: We could remove it and leave the top-level `linkWrap` function

Comment (felangel) [reply packages/mason_logger/lib/src/mason_logger.dart:60]: That's my preference in an attempt to keep things simple if it's alright with you.

Comment (Luckey-Elijah) [reply packages/mason_logger/lib/src/mason_logger.dart:60]: Totally fine with me :)

File: packages/mason_logger/lib/src/link_wrap.dart [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/// Helper function for "wrapping" an optional message with link.
+String linkWrap(String? message, Uri uri) {
```
Comment (felangel) [inline packages/mason_logger/lib/src/link_wrap.dart:2]:
```suggestion
String link({required Uri uri, String? message}) {
```

wdyt?

Comment (Luckey-Elijah) [reply packages/mason_logger/lib/src/link_wrap.dart:2]: Looks good! I think that makes it more clean how you can use it

**Comment by Luckey-Elijah** [comment]
Ready for a second another look. just have the single `link` function for wrapping messages

**Comment by felangel** [comment]
> Ready for a second another look. just have the single `link` function for wrapping messages

Sweet, thanks! Will review/merge shortly 🎉

File: packages/mason_logger/lib/src/link.dart [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
-  String? message,
-}) {
-  const lead = '\x1B]8;;';
-  const trail = '\x1B\\';
+/// Wraps [uri] with an escape sequence so it's recognized as a hyperlink.
+/// An optional message can be used in place of the [uri].
+/// If no [message] is provided, the text content will be the full [uri].
+///
+/// ```dart
+/// final plainLink = link(Uri.parse('https://dart.dev'));
```
Comment (Luckey-Elijah) [inline packages/mason_logger/lib/src/link.dart:6]: should be 
```
/// final plainLink = link(uri: Uri.parse('https://dart.dev'));
```

**Review by felangel — APPROVED** [review]
LGTM thanks so much for the contribution! 💯

**Pull request merged**